### PR TITLE
Isope2023 scrutineering review

### PR DESCRIPTION
**Pull request by ahasankhani**

### CODE REVIEW

File: modules.py [hunk 1]
```diff
@@ -14,56 +12,39 @@ def obj_terms(x_in, x_name, p_in: dict):
 
     # run each module 
     wave_out = wave_climate(wec,wave_in)
```
Comment (oliviavitale35) [inline modules.py:14]: for these commands you're using (wave_climate and fish_yield_func), are they built into the library? or did you write them somewhere else?

File: modules.py [hunk 2]
```diff
@@ -143,10 +109,9 @@ def P_rated(x_in, x_name, p_in: dict):
 def carrying_capacity_print(x_in, x_name, p_in: dict):
```
Comment (oliviavitale35) [inline modules.py:112]: i've noticed this quite a few times throughout this code, but i've noticed several functions don't have return statements. i'm not a python-pro, so i don't know if there's a specific reason for this, but i thought no return statement = nothing is returned from the function ?

File: run_sim_wecpen.ipynb [hunk 3]
```diff
@@ -0,0 +1,379 @@
+{
```
Comment (oliviavitale35) [inline run_sim_wecpen.ipynb:1]: i think this script gets really hard to follow because the commands mostly aren't labeled. might be nice to add some check-in #'s to describe what's going on

File: modules.py [hunk 4]
```diff
@@ -261,42 +223,44 @@ def default_values(var_category_names):
...
     
     if any('p_wec' in i for i in var_category_names):
         vals['wec_unit_cost'] = (0.45 * 1.19, '[$/kWh]')   # 'point absorber' * inflation rate from 2014 to 2022
-        vals['pen_unit_cost'] = (100, '[$/m^3]')    # 80 $/m^3 for net pen + 20 $/m^3 for mooring
-        vals['permeability'] = (0.8, '[-]')
         vals['capture_width_ratio_dict'] = (dict(zip(wec_types[0], capture_width_ratios[0])), '[-]')
         vals['wave_damping_dict'] = (dict(zip(wec_types[0], wave_dampings[0])), '[-]')
         
     if any('p_fish_salmon' in i for i in var_category_names):
         vals['F_p'] = (0.45, '[-]')
```
Comment (ommurphy) [inline modules.py]: Consider relisting the definitions for each of these variables so that you don't have to scroll back and forth if you forget. 
```suggestion
        vals['F_p'] = (0.45, '[-]')
```

File: README.md [hunk 5]
```diff
@@ -1,18 +1,38 @@
...
 Sensitivity Analysis: In order to run the "sensitivity analysis", you can open and run the param_sweep (.ipynb) file.
 
-In order to run the "Wave-powered Aquaculture" optimization code and find optimal wave energy convereter (WEC) and optimal pen considering constant environmental parameters, you can open and run the run_sim_wec&pen (.ipynb) file.
+In order to run the "Wave-powered Aquaculture" optimization code and find optimal wave energy convereter (WEC) and optimal pen considering constant environmental parameters, you can open and run the run_sim_wecpen (.ipynb) file.
+
+In order to run the "Wave-powered Aquaculture" optimization code and find optimal WEC and optimal pen with random values for starting points of design variables to check the convergence of the optimization, you can open and run the run_sim_wecpen_random_init (.ipynb) file.
+
+## Dependencies
+Python 3.8.8 is used to develop this package. 
+For other required libraries, please refer to requirements.txt file.
```
Comment (ommurphy) [inline README.md:23]: Consider adding the requirements.txt to the list in lines 6-11
```suggestion
For other required libraries, please refer to requirements.txt file.
```

File: modules.py [hunk 6]
```diff
@@ -14,56 +12,39 @@ def obj_terms(x_in, x_name, p_in: dict):
...
-    env_salinitymin_cons = pen.salinity - fish.salinity_min
-    env_salinitymax_cons = fish.salinity_max - pen.salinity
-    env_O2_min_cons = pen.O2_in - fish.O2_min
-    
-    #print('cons:', P_gen_cons, fish_yield_cons)
+    P_gen_cons = (wec.annual_energy - pen.power) / wec.annual_energy
 
+    fish_yield_cons = (carrying_capacity - fish_yield) / carrying_capacity
+ 
+    pen_ratio_low_cons = (pen.D - pen.H) / pen.D
```
Comment (ommurphy) [inline modules.py:41]: Consider explaining what these ratios are for
```suggestion
    pen_ratio_low_cons = (pen.D - pen.H) / pen.D
```

File: objects.py [hunk 7]
```diff
@@ -263,25 +274,14 @@ def power(self) -> float:
 
     
     def carrying_capacity(self, fish) -> float:
-        length = self.n * self.D   # Based on worst case
-        #length = self.n * self.D + self.spacing * (self.n-1) # From reference paper for a row farm
-        #length = self.D # for each pen
-
-        #print('length',length)
+        #length = self.D   # Based on worst case
+        length = self.n * self.D + self.spacing * (self.n-1) # From reference paper for a row farm
```
Comment (ommurphy) [inline objects.py]: Consider providing abbreviated citation for the reference paper here 
```suggestion
        length = self.n * self.D + self.spacing * (self.n-1) # From reference paper for a row farm
```

File: run_sim_wecpen.ipynb (hunk 3 above)
Comment (ommurphy) [reply run_sim_wecpen.ipynb:1]: +1

File: modules.py (hunk 1 above)
Comment (ahasankhani) [reply modules.py:14]: They've been written in the modules.py file.

File: modules.py (hunk 6 above)
Comment (ahasankhani) [reply modules.py:41]: added explanation as a comment to code

File: modules.py (hunk 2 above)
Comment (ahasankhani) [reply modules.py:112]: they've been used to print some values and don't return any results.

File: modules.py (hunk 4 above)
Comment (ahasankhani) [reply modules.py]: updated.

File: objects.py (hunk 7 above)
Comment (ahasankhani) [reply objects.py]: updated.

File: run_sim_wecpen.ipynb (hunk 3 above)
Comment (ahasankhani) [reply run_sim_wecpen.ipynb:1]: Updated.

File: README.md (hunk 5 above)
Comment (ahasankhani) [reply README.md:23]: updated.
